core/commands/object.go

```diff
@@ -701,6 +702,24 @@ func objectPut(n *core.IpfsNode, input io.Reader, encoding string) (*Object, err
 	case objectEncodingProtobuf:
 		dagnode, err = dag.Decoded(data)
 
+	case objectEncodingXML:
+		node := new(Node)
+		err = xml.Unmarshal(data, node)
+		if err != nil {
+			return nil, err
+		}
+
+		// check that we have data in the Node to add
+		// otherwise we will add the empty object without raising an error
+		if NodeEmpty(node) {
+			return nil, ErrEmptyNode
+		}
```

Comment:
the empty node is a valid object. it may be the result of a script. i think this should only be an error with an option like `--no-empty`, but that by default it should be ok. what is the current behavior with protobuf and json?

cc @whyrusleeping